Common/cpp/Registries/EventHandlerRegistry.cpp

```diff
@@ -22,8 +22,18 @@ void EventHandlerRegistry::unregisterEventHandler(unsigned long id) {
 void EventHandlerRegistry::processEvent(jsi::Runtime &rt, std::string eventName, std::string eventPayload) {
   auto handlersIt = eventMappings.find(eventName);
   if (handlersIt != eventMappings.end()) {
-    // TODO: use jsi::Value::createFromJsonUtf8
-    auto eventObject = eval(rt, ("(" + eventPayload + ")").c_str()).asObject(rt).getProperty(rt, "NativeMap");
+    // We receive here a JS Map with JSON as a value of NativeMap key
+    // { NativeMap: { "jsonProp": "json value" } }
+    // So we need to extract only JSON part
+    std::string delimimter = "NativeMap:";
+    auto positionToSplit = eventPayload.find(delimimter) + delimimter.size();
+    auto lastBracketCharactedPosition = eventPayload.size() - positionToSplit - 1;
+    auto eventJSON = eventPayload.substr(positionToSplit,  lastBracketCharactedPosition);
+
+    std::vector<uint8_t> eventJSONVector(eventJSON.begin(), eventJSON.end());
```

Comment:
Creating a vector here seems to be unnecessary. Simple conversion in the next instruction works:
`auto eventObject = jsi::Value::createFromJsonUtf8(rt, (uint8_t*)(&eventJSON[0]), eventJSON.size());`

Reply:
Yep, make sense, didn't know how to convert it properly here. One more thing learned today, thanks :)